### PR TITLE
client: use next ssh key on HTTP_FORBIDDEN

### PR DESCRIPTION
**Pull request by davidxia**

### CODE REVIEW

**Comment by mattnworb** [comment]
does #728 replace this?

**Comment by davidxia** [comment]
@mattnworb No the two are unrelated. I havent looked into why tests fail on this one yet.

**Comment by davidxia** [comment]
@rohansingh

File: helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java [hunk 1]
```diff
@@ -243,7 +243,7 @@ private HttpURLConnection connect(final URI uri, final String method, final byte
 
               final int responseCode = connection.getResponseCode();
               if (((responseCode == HTTP_FORBIDDEN) || (responseCode == HTTP_UNAUTHORIZED))
-                  && !identities.isEmpty()) {
+                  && connection.getHeaderField("WWW-Authenticate") != null) {
```
Comment (davidxia) [inline helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java]: @rohansingh We need some way to differentiate between app-level and TLS-level unauthed responses. Would a header set by nginx work?

**Comment by davidxia** [comment]
Replaced by https://github.com/spotify/helios/pull/749

**Pull request closed without merging**